src/app/repositories/firebase/firebase_loader.py

```diff
@@ -0,0 +1,42 @@
+from config.config_factory import ConfigFactory
+from google.cloud import firestore
+
+config = ConfigFactory.get_config()
+
+
+class FirebaseLoader:
+    def __init__(self):
+        self.database = None
+        self.datasets_collection = None
+        self.schemas_collection = None
+        self._connect_db()
+        self._set_collections()
```

Comment:
Feels a bit weird that you're setting these values to None, then immediately setting them to actual values in `_connect_db` and `_set_collections`. Why not have this:

```
def __init__(self):
    self.database = self.connect_db()
    self.datasets_collection = self.set_collection('datasets')
    self.schemas_collection = self.set_collection('schemas')
```

`_set_collections` would change to `set_collection`, where you pass it the collection name as a parameter.